### PR TITLE
17207 - Alterations - add in new payment code for BC -> ULC

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:* /bcgov/entity#17207

*Description of changes:*
Also includes fixes from previous PR's. 

Relies on:
https://github.com/bcgov/bcrs-shared-components/pull/190/files

and a new bcrs-shared enums to be pushed out -^

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by seeker25** [comment]
image

File: src/views/Alteration.vue [hunk 1]
```diff
@@ -215,6 +215,30 @@ export default class Alteration extends Mixins(CommonMixin, FeeMixin, FilingTemp
...
 
+  @Watch('hasBusinessTypeChanged')
+  onBusinessTypeChanged () {
+    const filingData = this.getFilingData
+    // When we are converting from BC to ULC, it's $1000 not $100.
+    if (this.getEntitySnapshot?.businessInfo?.legalType === CorpTypeCd.BC_COMPANY &&
+        this.getEntityType === CorpTypeCd.BC_ULC_COMPANY) {
+      // Travis Semple - will use ENUM after bcrs-shared-components has been updated.
+      this.setFilingData([{
+        filingTypeCode: 'NOALU' as FilingCodes,
```
Comment (seeker25) [inline src/views/Alteration.vue:226]: Will move this back

File: tests/unit/Alteration.spec.ts [hunk 2]
```diff
@@ -397,6 +398,21 @@ describe('Alteration component', () => {
...
+    store.stateModel.tombstone.entityType = CorpTypeCd.BC_ULC_COMPANY
+    store.stateModel.entitySnapshot = {
+      businessInfo: {
+        legalType: CorpTypeCd.BC_COMPANY
+      }
+    } as EntitySnapshotIF
+    await Vue.nextTick()
+
+    // Travis Semple - will use ENUM after bcrs-shared-components has been updated.
+    expect(store.stateModel.filingData[0].filingTypeCode).toBe('NOALU')
```
Comment (seeker25) [inline tests/unit/Alteration.spec.ts:413]: Will move this back

**Comment by pwei1018** [comment]
Temporary Url for review: https://business-edit-dev--pr-521-c5zdlcko.web.app

File: src/views/Alteration.vue (hunk 1 above)
Comment (severinbeauvais) [inline src/views/Alteration.vue:226]: Is there an easy way to use different filing data (in a resource file?) so we don't need a special case here?

**Review by severinbeauvais — COMMENTED** [review]
Thanks for the test util cleanup.

I'll look at this again after the new pay code is published and you've updated the code here.

**Review by Jxio — COMMENTED** [review]
looks good to me

**Review by Jxio — APPROVED** [review]

**Review by severinbeauvais — APPROVED** [review]

**Comment by pwei1018** [comment]
Temporary Url for review: https://business-edit-dev--pr-521-c5zdlcko.web.app

**Pull request merged**